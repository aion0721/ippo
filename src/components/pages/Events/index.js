import React from 'react';
import styled from 'styled-components';
import propTypes from 'prop-types';
import dayjs from 'dayjs';
import { Spinner, Text } from '@blueprintjs/core';
import NavigationBar from 'src/components/molecules/NavigationBar';
import EventCard from 'src/components/organisms/EventCard';
import SearchForm from 'src/components/organisms/SerchForm';

const Container = styled.div`
  padding: 10px 15px;
`;

const EventCardContainer = styled.div`
  margin: 8px 0;
`;

const Events = ({ data: { loading, connpass } }) => (
  <>
    <NavigationBar appName="IPPO" />
    <Container>
      {loading ? (
        <Spinner />
      ) : connpass ? (
        <>
          <SearchForm />
          {connpass.events.map(event => (
            <EventCardContainer key={event.event_id}>
              <EventCard
                title={event.title}
                eventUrl={event.event_url}
                catchMessage={event.catch}
                place={event.place}
                datetime={
                  event.started_at && dayjs(event.started_at).format('YYYY年MM月DD日 HH:mm〜')
                }
                interactive
              />
            </EventCardContainer>
          ))}
        </>
      ) : (
        <Text>No Contents</Text>
      )}
    </Container>
  </>
);

Events.displayName = 'Events';

Events.propTypes = {
  data: propTypes.shape({
    loading: propTypes.bool.isRequired,
    connpass: propTypes.shape({
      events: propTypes.arrayOf(
        propTypes.shape({
          title: propTypes.string,
          event_url: propTypes.string,
          catch: propTypes.string,
          place: propTypes.string,
          started_at: propTypes.string,
        }),
      ),
    }),
  }),
};

Events.defaultProps = {
  data: {
    loading: false,
    connpass: {
      events: [],
    },
  },
};

export default Events;
