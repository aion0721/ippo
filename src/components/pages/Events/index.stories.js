import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import Events from '.';

const stories = storiesOf('pages/Events', module);

const event = i => ({
  event_id: i,
  title: `イベントのタイトル${i}`,
  event_url: 'https://connpass.com/',
  catch: 'イベントの説明です',
  place: '東京都千代田区丸の内',
  started_at: '2012-04-17T18:30:00+09:00',
});

const data = {
  connpass: {
    events: [...new Array(10)].map((_, i) => event(i)),
  },
};

stories.add('通常パターン', () => (
  <Events data={data} registerNotification={action('registerNotification')} />
));

stories.add('ロード中', () => (
  <Events data={{ loading: true }} registerNotification={action('registerNotification')} />
));

stories.add('データなし', () => (
  <Events data={{ events: null }} registerNotification={action('registerNotification')} />
));
