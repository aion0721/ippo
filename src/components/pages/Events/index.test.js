import React from 'react';
import { snapshot } from 'test/helpers';
import Events from '.';

const event = i => ({
  event_id: i,
  title: `イベントのタイトル${i}`,
  event_url: 'https://connpass.com/',
  catch: 'イベントの説明です',
  place: '東京都千代田区丸の内',
  started_at: '2012-04-17T18:30:00+09:00',
});

const events = {
  connpass: {
    events: [...new Array(10)].map((_, i) => event(i)),
    results_available: 10,
  },
};

const props = ({ loading = false, events = null }) => ({
  data: {
    loading,
    events,
    refetch: jest.fn(),
  },
});

snapshot('Events/nomal', <Events {...props({ events })} />);

snapshot('Events/loading', <Events {...props({ loading: true })} />);

snapshot('Events/nocontents', <Events {...props({})} />);
