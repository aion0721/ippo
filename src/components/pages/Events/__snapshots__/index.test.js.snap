// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: Events/loading 1`] = `
.c0 {
  padding: 10px 15px;
}

<Events
  data={
    Object {
      "events": null,
      "loading": true,
      "refetch": [MockFunction],
    }
  }
  registerNotification={[MockFunction]}
>
  <NavigationBar
    appName="IPPO"
  >
    <Blueprint3.Navbar>
      <div
        className="bp3-navbar"
      >
        <Blueprint3.NavbarGroup
          align="left"
        >
          <div
            className="bp3-navbar-group bp3-align-left"
          >
            <Blueprint3.NavbarHeading>
              <div
                className="bp3-navbar-heading"
              >
                IPPO
              </div>
            </Blueprint3.NavbarHeading>
          </div>
        </Blueprint3.NavbarGroup>
      </div>
    </Blueprint3.Navbar>
  </NavigationBar>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  padding: 10px 15px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        <Blueprint3.Spinner>
          <div
            className="bp3-spinner"
          >
            <div
              className="bp3-spinner-animation"
            >
              <svg
                height={50}
                strokeWidth="8.00"
                viewBox="1.00 1.00 98.00 98.00"
                width={50}
              >
                <path
                  className="bp3-spinner-track"
                  d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                />
                <path
                  className="bp3-spinner-head"
                  d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                  pathLength={280}
                  strokeDasharray="280 280"
                  strokeDashoffset={210}
                />
              </svg>
            </div>
          </div>
        </Blueprint3.Spinner>
      </div>
    </StyledComponent>
  </styled.div>
</Events>
`;

exports[`should match snapshot: Events/nocontents 1`] = `
.c0 {
  padding: 10px 15px;
}

<Events
  data={
    Object {
      "events": null,
      "loading": false,
      "refetch": [MockFunction],
    }
  }
  registerNotification={[MockFunction]}
>
  <NavigationBar
    appName="IPPO"
  >
    <Blueprint3.Navbar>
      <div
        className="bp3-navbar"
      >
        <Blueprint3.NavbarGroup
          align="left"
        >
          <div
            className="bp3-navbar-group bp3-align-left"
          >
            <Blueprint3.NavbarHeading>
              <div
                className="bp3-navbar-heading"
              >
                IPPO
              </div>
            </Blueprint3.NavbarHeading>
          </div>
        </Blueprint3.NavbarGroup>
      </div>
    </Blueprint3.Navbar>
  </NavigationBar>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  padding: 10px 15px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        <Blueprint3.Text>
          <div
            className=""
          >
            No Contents
          </div>
        </Blueprint3.Text>
      </div>
    </StyledComponent>
  </styled.div>
</Events>
`;

exports[`should match snapshot: Events/nomal 1`] = `
.c0 {
  padding: 10px 15px;
}

<Events
  data={
    Object {
      "events": Object {
        "connpass": Object {
          "events": Array [
            Object {
              "catch": "イベントの説明です",
              "event_id": 0,
              "event_url": "https://connpass.com/",
              "place": "東京都千代田区丸の内",
              "started_at": "2012-04-17T18:30:00+09:00",
              "title": "イベントのタイトル0",
            },
            Object {
              "catch": "イベントの説明です",
              "event_id": 1,
              "event_url": "https://connpass.com/",
              "place": "東京都千代田区丸の内",
              "started_at": "2012-04-17T18:30:00+09:00",
              "title": "イベントのタイトル1",
            },
            Object {
              "catch": "イベントの説明です",
              "event_id": 2,
              "event_url": "https://connpass.com/",
              "place": "東京都千代田区丸の内",
              "started_at": "2012-04-17T18:30:00+09:00",
              "title": "イベントのタイトル2",
            },
            Object {
              "catch": "イベントの説明です",
              "event_id": 3,
              "event_url": "https://connpass.com/",
              "place": "東京都千代田区丸の内",
              "started_at": "2012-04-17T18:30:00+09:00",
              "title": "イベントのタイトル3",
            },
            Object {
              "catch": "イベントの説明です",
              "event_id": 4,
              "event_url": "https://connpass.com/",
              "place": "東京都千代田区丸の内",
              "started_at": "2012-04-17T18:30:00+09:00",
              "title": "イベントのタイトル4",
            },
            Object {
              "catch": "イベントの説明です",
              "event_id": 5,
              "event_url": "https://connpass.com/",
              "place": "東京都千代田区丸の内",
              "started_at": "2012-04-17T18:30:00+09:00",
              "title": "イベントのタイトル5",
            },
            Object {
              "catch": "イベントの説明です",
              "event_id": 6,
              "event_url": "https://connpass.com/",
              "place": "東京都千代田区丸の内",
              "started_at": "2012-04-17T18:30:00+09:00",
              "title": "イベントのタイトル6",
            },
            Object {
              "catch": "イベントの説明です",
              "event_id": 7,
              "event_url": "https://connpass.com/",
              "place": "東京都千代田区丸の内",
              "started_at": "2012-04-17T18:30:00+09:00",
              "title": "イベントのタイトル7",
            },
            Object {
              "catch": "イベントの説明です",
              "event_id": 8,
              "event_url": "https://connpass.com/",
              "place": "東京都千代田区丸の内",
              "started_at": "2012-04-17T18:30:00+09:00",
              "title": "イベントのタイトル8",
            },
            Object {
              "catch": "イベントの説明です",
              "event_id": 9,
              "event_url": "https://connpass.com/",
              "place": "東京都千代田区丸の内",
              "started_at": "2012-04-17T18:30:00+09:00",
              "title": "イベントのタイトル9",
            },
          ],
          "results_available": 10,
        },
      },
      "loading": false,
      "refetch": [MockFunction],
    }
  }
  registerNotification={[MockFunction]}
>
  <NavigationBar
    appName="IPPO"
  >
    <Blueprint3.Navbar>
      <div
        className="bp3-navbar"
      >
        <Blueprint3.NavbarGroup
          align="left"
        >
          <div
            className="bp3-navbar-group bp3-align-left"
          >
            <Blueprint3.NavbarHeading>
              <div
                className="bp3-navbar-heading"
              >
                IPPO
              </div>
            </Blueprint3.NavbarHeading>
          </div>
        </Blueprint3.NavbarGroup>
      </div>
    </Blueprint3.Navbar>
  </NavigationBar>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  padding: 10px 15px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        <Blueprint3.Text>
          <div
            className=""
          >
            No Contents
          </div>
        </Blueprint3.Text>
      </div>
    </StyledComponent>
  </styled.div>
</Events>
`;
